### PR TITLE
no-issue: Add free-threading CI

### PR DESCRIPTION
**Pull request by corona10**
reference: https://dev.to/hugovk/help-us-test-free-threaded-python-without-the-gil-1hgf

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -25,44 +25,54 @@ jobs:
...
         include:
           # As the experimental task for the dev version.
           - os: ubuntu-latest
             python: "3.13-dev"
             experimental: true
+            build: ''
+          - os: ubuntu-latest
+            python: "3.13-dev"
+            experimental: true
+            build: 'free_threading'
```
Comment (hugovk) [inline .github/workflows/main.yml:38]: To match line 67:
```suggestion
            build: 'free-threading'
```

**Comment by corona10** [comment]
@hugovk 
Thanks, now you can review the changes :)

**Comment by corona10** [comment]
Sorry my local force push up ignore your previous comment :(

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -25,44 +25,54 @@ jobs:
         os: [ubuntu-latest]
         python: ["3.8", "3.9", "3.10", "3.11", "3.12"]
         experimental: [false]
+        build: ['']
```
Comment (corona10) [inline .github/workflows/main.yml:28]: OTAH, what do you think about using 'default'?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -25,44 +25,54 @@ jobs:
...
+        nogil: ${{ matrix.build == 'free-threading' }}
     - name: Install
       # pyperformance must be installed:
       # pyperformance/tests/test_compare.py imports it
       run: |
         python -m pip install --upgrade pip setuptools
         python -m pip install -e .
     - name: Display Python version
-      run: python -c "import sys; print(sys.version)"
+      run: python --version --version
```
Comment (hugovk) [inline .github/workflows/main.yml:75]: It's a shame this doesn't show if we're on a free-threaded build, but never mind.

File: .github/workflows/main.yml (hunk 2 above)
Comment (hugovk) [reply .github/workflows/main.yml:28]: Yeah, that could work. I've no strong opinion either way :)

**Review by hugovk — APPROVED** [review]

**Pull request merged**